lib/src/main/java/com/scalar/admin/kubernetes/Pauser.java

```diff
@@ -84,17 +85,30 @@ public PausedDuration pause(int pauseDuration, @Nullable Long maxPauseWaitTime)
       throw new PauserException("Failed to find the target pods to pause.", e);
     }
 
-    RequestCoordinator coordinator = getRequestCoordinator(target);
+    RequestCoordinator coordinator;
+    try {
+      coordinator = getRequestCoordinator(target);
+    } catch (Exception e) {
+      throw new PauserException("Failed to initialize the coordinator.", e);
+    }
+
+    Instant startTime;
+    Instant endTime;
+    try {
+      coordinator.pause(true, maxPauseWaitTime);
 
-    coordinator.pause(true, maxPauseWaitTime);
+      startTime = Instant.now();
 
-    Instant startTime = Instant.now();
+      Uninterruptibles.sleepUninterruptibly(pauseDuration, TimeUnit.MILLISECONDS);
 
-    Uninterruptibles.sleepUninterruptibly(pauseDuration, TimeUnit.MILLISECONDS);
+      endTime = Instant.now();
 
-    Instant endTime = Instant.now();
+      unpauseWithRetry(coordinator, MAX_UNPAUSE_RETRY_COUNT, target);
 
-    unpauseWithRetry(coordinator, MAX_UNPAUSE_RETRY_COUNT, target);
+    } catch (Exception e) {
+      unpauseWithRetry(coordinator, MAX_UNPAUSE_RETRY_COUNT, target);
+      throw e;
+    }
```

Comment:
This is the main update of this PR.

Previously, there was no exception handling. So, there is a possibility that the Scalar products pod will keep `paused` status forever. In such a case, the Scalar product does not accept any request. It causes the service down.

To avoid such a case as long as possible, we want to run `unpause()` if any errors occur in any parts of the process.